src/connectors/current-refinements/connectCurrentRefinements.js

```diff
@@ -0,0 +1,305 @@
+            refinements,
+            instantSearchInstance,
+            widgetParams,
+          },
+          false
+        );
+      },
+
+      dispose() {
+        unmountFn();
+      },
+    };
+  };
+}
+
+function getNormalizedRefinements({
+  results,
+  state,
+  helper,
+  includedAttributes,
+  excludedAttributes,
+  clearsQuery,
+}) {
+  const refinements = getRefinements(results, state, clearsQuery)
+    .filter(
+      refinement =>
+        !includedAttributes ||
+        includedAttributes.indexOf(refinement.attributeName) !== -1
+    )
+    .filter(
+      refinement => excludedAttributes.indexOf(refinement.attributeName) === -1
+    )
+    .map(normalizeRefinementItem);
+
+  return normalizeRefinements(refinements, helper);
+}
+
+function clearRefinementFromState(state, refinementItem) {
+  switch (refinementItem.type) {
+    case 'facet':
+      return state.removeFacetRefinement(
+        refinementItem.attribute,
+        refinementItem.value
+      );
+    case 'disjunctive':
+      return state.removeDisjunctiveFacetRefinement(
+        refinementItem.attribute,
+        refinementItem.value
+      );
+    case 'hierarchical':
+      return state.clearRefinements(refinementItem.attribute);
+    case 'exclude':
+      return state.removeExcludeRefinement(
+        refinementItem.attribute,
+        refinementItem.value
+      );
+    case 'numeric':
+      return state.removeNumericRefinement(
+        refinementItem.attribute,
+        refinementItem.operator,
+        refinementItem.value
+      );
+    case 'tag':
+      return state.removeTagRefinement(refinementItem.value);
+    case 'query':
+      return state.setQueryParameter('query', '');
+    default:
+      throw new Error(
+        `clearRefinement: type ${refinementItem.type} is not handled`
+      );
+  }
+}
+
+function clearRefinement(helper, refinement) {
+  helper.setState(clearRefinementFromState(helper.state, refinement)).search();
+}
+
+function getOperatorSymbol(operator) {
+  switch (operator) {
+    case '>=':
+      return '≥';
+    case '<=':
+      return '≤';
+    default:
+      return '';
+  }
+}
+
+function normalizeRefinementItem(item) {
+  const attribute = item.type === 'query' ? 'query' : item.attributeName;
+  const label = item.operator
+    ? `${getOperatorSymbol(item.operator)} ${item.name}`
+    : item.name;
+  const value = item.type === 'numeric' ? Number(item.name) : item.name;
+
+  return {
+    attribute,
+    type: item.type,
+    value,
+    label: item.type === 'query' ? `"${label}"` : label,
+    ...(item.operator && { operator: item.operator }),
+    ...(item.count && { count: item.count }),
+    ...(item.exhaustive && { exhaustive: item.exhaustive }),
+  };
+}
+
+function compareObjects(a, b, attribute) {
+  if (a[attribute] === b[attribute]) {
+    return 0;
+  }
+
+  if (a[attribute] < b[attribute]) {
+    return -1;
+  }
+
+  return 1;
+}
+
+function normalizeRefinements(refinements, helper) {
```

Comment:
Refinements are sorted alphabetically. Is that what we want?

Reply:
As I said, sorting is something we never specified so to me alpha makes more sense than the complicated logic it had before, but no sorting at all is also an option for me.